### PR TITLE
chore: Typescript Cleanup Part 3 of 3 for Visualization Type error

### PR DESCRIPTION
**Pull request by v-prachin**
#### Details

Typescript Cleanup Part 3 of 3 
We have total 90 errors remaining in 53 files.

##### Motivation
Visualization type was having error when negative value is assigned to it, so changed the datatype to number along with visualization type to overcome the errors. The test cases are passing successfully and able to observe in console log that the value of e.g test = -1 is coming -1 only even after changing the datatype for visualizationType.

1. src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts (5)
2. src/tests/unit/tests/injected/analyzers/analyzer-provider.test.ts (1)
3. src/tests/unit/tests/injected/analyzers/tab-stops-analyzer.test.ts (1)
4. src/tests/unit/tests/injected/selector-map-helper.test.ts (4)
5. src/tests/unit/tests/injected/target-page-visualization-updater.test.ts	(1)
6. src/tests/unit/tests/injected/visualization-state-change-handler.test.ts (2)
7. src/tests/unit/tests/injected/visualization-type-drawer-registrar.test.ts (1)
8. src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx (8)
9. src/tests/unit/tests/background/stores/visualization-scan-result-store.test.ts (2)

Thus, total number of errors covered as part of this user story: **25**

#### Pull request checklist

- [x] Addresses an existing issue: #(https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/injected/analyzers/analyzer.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export interface ScanBasePayload extends BaseActionPayload {
 
 export interface AnalyzerConfiguration {
     key: string;
-    testType: VisualizationType;
+    testType: VisualizationType | number;
```
Comment (v-viyada) [inline src/injected/analyzers/analyzer.ts]:
```suggestion
    testType: VisualizationType;
```

File: src/tests/unit/tests/background/stores/visualization-scan-result-store.test.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ describe('VisualizationScanResultStoreTest', () => {
     test('onRescanVisualization: for test that has state', async () => {
         const tabEvents: TabbedElementData[] = getTabbedElementDataStub();
         const testKey = AdHocTestkeys.TabStops;
-        const visualizationTypeStub = -2;
+        const visualizationTypeStub: number = -2;
```
Comment (v-viyada) [inline src/tests/unit/tests/background/stores/visualization-scan-result-store.test.ts]:
```suggestion
        const visualizationTypeStub= 0;
```

File: src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts [hunk 3]
```diff
@@ -93,8 +93,9 @@ describe('AnalyzerStateUpdateHandlerTest', () => {
         setupDefaultVisualizationConfigFactory();
         setupIsTestTerminated(configMock, requirementConfig, prevState, newState, false, true);
 
+        const type1: number = -1;
         testObject.handleUpdate(newState);
-        tearDownCallback(configMock.object, -1, requirementConfig);
+        tearDownCallback(configMock.object, type1, requirementConfig);
```
Comment (v-viyada) [inline src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts]:
```suggestion
        tearDownCallback(configMock.object, 0, requirementConfig);
```

File: src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts [hunk 4]
```diff
@@ -93,8 +93,9 @@ describe('AnalyzerStateUpdateHandlerTest', () => {
         setupDefaultVisualizationConfigFactory();
         setupIsTestTerminated(configMock, requirementConfig, prevState, newState, false, true);
 
+        const type1: number = -1;
```
Comment (v-viyada) [inline src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts]:
```suggestion
```

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx [hunk 5]
```diff
@@ -19,9 +19,9 @@ import { ShortcutCommandsTestData } from '../../../common/sample-test-data';
 import { VisualizationStoreDataBuilder } from '../../../common/visualization-store-data-builder';
 
 describe('DiagnosticViewToggleFactoryTest', () => {
-    const firstVisualizationTypeStub = -1;
-    const secondVisualizationTypeStub = -2;
-    const thirdAlwaysDisabledVisualizationTypeStub = -3;
+    const firstVisualizationTypeStub: number = -1;
```
Comment (v-viyada) [inline src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]:
```suggestion
    const firstVisualizationTypeStub: number = 0;
```

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx [hunk 6]
```diff
@@ -19,9 +19,9 @@ import { ShortcutCommandsTestData } from '../../../common/sample-test-data';
...
+    const secondVisualizationTypeStub: number = -2;
```
Comment (v-viyada) [inline src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]:
```suggestion
    const secondVisualizationTypeStub: number = 1;
```

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx [hunk 7]
```diff
@@ -19,9 +19,9 @@ import { ShortcutCommandsTestData } from '../../../common/sample-test-data';
...
+    const thirdAlwaysDisabledVisualizationTypeStub: number = -3;
```
Comment (v-viyada) [inline src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]:
```suggestion
    const thirdAlwaysDisabledVisualizationTypeStub: number = 2;
```

**Review by v-viyada — CHANGES_REQUESTED** [review]

File: src/injected/analyzers/analyzer.ts (hunk 1 above)
Comment (v-prachin) [reply src/injected/analyzers/analyzer.ts]: Made the suggested change

File: src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts (hunk 3 above)
Comment (v-prachin) [reply src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts]: Made the suggested change

File: src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts (hunk 4 above)
Comment (v-prachin) [reply src/tests/unit/tests/injected/analyzer-state-update-handler.test.ts]: Made the suggested change

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx (hunk 5 above)
Comment (v-prachin) [reply src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]: Made the suggested change

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx (hunk 6 above)
Comment (v-prachin) [reply src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]: Made the suggested change

File: src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx (hunk 7 above)
Comment (v-prachin) [reply src/tests/unit/tests/popup/components/diagnostic-view-toggle-factory.test.tsx]: Made the suggested change

**Pull request merged**